src/jormungandr/explorer/src/api/graphql/mod.rs

```diff
@@ -1398,6 +1394,41 @@ impl VotePlanStatus {
     }
 }
 
+/// checks if the tally is public or private and converts result to TallyStatus type
+pub fn to_tally_status(tally: Option<ExplorerVoteTally>) -> Option<TallyStatus> {
+    tally.map(|tally| match tally {
+        ExplorerVoteTally::Public { results, options } => TallyStatus::Public(TallyPublicStatus {
+            results: results.iter().map(Into::into).collect(),
+            options: options.into(),
+        }),
+        ExplorerVoteTally::Private { results, options } => {
+            TallyStatus::Private(TallyPrivateStatus {
+                results: results.map(|res| res.iter().map(Into::into).collect()),
+                options: options.into(),
+            })
+        }
+    })
+}
```

Comment:
I believe this could be a `impl From ExplorerVoteTally for TallyStatus` removing the inner `Option::map` which can be done outside this fn